### PR TITLE
Update top app bar to latest

### PR DESCRIPTION
**Pull request by williamernest**
Updates the top app bar package to the latest.

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -45,9 +45,11 @@ class App extends Component {
     return (
       <div>
         <HeaderBar isTopPage />
-        <ul id='catalog-image-list' className='mdc-image-list standard-image-list'>
-          {this.renderListItem('Button', buttonImg, '/material-components-web-catalog/button')}
-        </ul>
+        <div className='mdc-top-app-bar--fixed-adjust'>
+          <ul id='catalog-image-list' className='mdc-image-list standard-image-list'>
```
Comment (bonniezhou) [inline src/App.js]: Is it possible to move the `mdc-top-app-bar--fixed-adjust` class to the `<ul>` element?

File: src/ComponentPage.js [hunk 2]
```diff
@@ -64,10 +64,12 @@ class ComponentPage extends Component {
 
   render() {
     return (
-      <div className='mdc-layout-grid'>
-        <div className='mdc-layout-grid__inner'>
-          {this.renderSidebar(this.props.title)}
-          {this.renderDemoWrapper()}
+      <div className='mdc-top-app-bar--fixed-adjust'>
+        <div className='mdc-layout-grid '>
```
Comment (bonniezhou) [inline src/ComponentPage.js]: Remove extra space at the end.

File: src/HeaderBar.js [hunk 3]
```diff
@@ -1,21 +1,29 @@
...
+
+  render () {
+    return (
+        <header className='mdc-top-app-bar catalog-top-app-bar' ref='demoTopAppBar'>
+          <div className='mdc-top-app-bar__row'>
+            <section
+                className='mdc-top-app-bar__section mdc-top-app-bar__section--align-start'>
+              <HeaderIcon isTopPage={this.props.isTopPage}/>
+              <span className='mdc-top-app-bar__title catalog-top-app-bar__title'>
             Material Components for Web
```
Comment (bonniezhou) [inline src/HeaderBar.js]: Indent these 2 lines

File: src/styles/App.scss [hunk 4]
```diff
@@ -46,3 +46,10 @@ body {
   width: 80px;
   margin: auto;
 }
+
+// Needed until the following PR gets resolved
+// https://github.com/material-components/material-components-web/pull/2643
+.mdc-top-app-bar--fixed-adjust {
```
Comment (bonniezhou) [inline src/styles/App.scss]: Move to HeaderBar.js?

File: src/styles/HeaderBar.scss [hunk 5]
```diff
@@ -4,6 +4,7 @@
 
 .catalog-top-app-bar {
   @include mdc-top-app-bar-fill-color($material-color-grey-900);
+  z-index: 2;
```
Comment (bonniezhou) [inline src/styles/HeaderBar.scss]: Is this still necessary with the padding-top and margin adjusted?

Comment (williamernest) [reply src/styles/HeaderBar.scss]: Yes, the buttons end up above the top app bar because of stacking context.

**Pull request merged**